extract.R

```diff
@@ -0,0 +1,173 @@
+semtext <- function(fit, x2=T, cfi=F, tli=T, rmsea=T, rmsea.ci=T, srmr=T, bic=F, aic=F,
+                    clipboard=T, type = "scaled", srmr.type = "srmr", output="text",
+                    modelnames=NULL,
+                    sep = "\t", dec=",") {
+  ## fitted model or list of fitted model
+  ## type: standard , scaled , robust
+  ## srmr.type: srmr, srmr_bentler, srmr_bentler_nomean, srmr_bollen, srmr_bollen_nomean, srmr_mplus, srmr_mplus_nomean
+  
+  if(!is.null(modelnames)) {
+    model.names <- modelnames
+  } else {
+    model.names <- as.list(sys.call())[[2]]
+    model.names <- gsub("list", "", model.names)[-1]
+  }
```

Comment:
Co maji radky 9-14 presne delat?

Reply:
Vytáhnou jména modelů z toho listu, který jsi do modelu vložil, ale odstraňují to list a ponechávají jenom samotné názvy proměnných původních objektů. Nevím, jak to udělat lépe.

Reply:
@hynekcigler Spis jsem asi potreboval vysvetlit nac presne je tam ten `sys.call()` :)